### ISSUE TITLE
Connection problem with WLST script.

### ISSUE THREAD

**Issue by antonionappi88** [oracle/weblogic-kubernetes-operator#70]
Hello,
I'm not able to connect to the Admin using a wlst script. On the kubernetes side it is correctly configured : 

```kubectl
NAME                                        TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)             AGE
domain1-admin-server                        NodePort    10.21.64.58     <none>        7001:30701/TCP      3h
domain1-admin-server-extchannel-t3channel   NodePort    10.21.172.82    <none>        30012:30012/TCP     3h
```

I tried from my pc  and I have this : 

```python
wls:/offline> connect('weblogic','welcome1','t3://PUBLIC_IP:30012')
Connecting to t3://PUBLIC_IP:30012 with userid weblogic ...
Traceback (innermost last):
  File "<console>", line 1, in ?
  File "<iostream>", line 19, in connect
  File "<iostream>", line 553, in raiseWLSTException
WLSTException: Error occurred while performing connect : Error getting the initial context. There is no server running at t3://PUBLIC_IP:30012 : Timed out while attempting to establish connection to :t3://PUBLIC_IP:30012 
Use dumpStack() to view the full stacktrace :
```
or 

```python
wls:/offline> connect('weblogic','welcome1','http://PUBLIC_IP:30701')
Connecting to http://PUBLIC_IP:30701 with userid weblogic ...
Traceback (innermost last):
  File "<console>", line 1, in ?
  File "<iostream>", line 19, in connect
  File "<iostream>", line 553, in raiseWLSTException
WLSTException: Error occurred while performing connect : Error getting the initial context. There is no server running at http://PUBLIC_IP:30701 : http://PUBLIC_IP:30701: [RJVM:000575]Destination PUBLIC_IP, 30701 unreachable.; nested exception is: 
        java.io.IOException: Could not connect to http://PUBLIC_IP:30701; [RJVM:000576]No available router to destination.; nested exception is: 
        java.rmi.ConnectException: [RJVM:000576]No available router to destination. 
Use dumpStack() to view the full stacktrace :
```
Tcpdump on my pc and on the OCI servers seems fine (there are packets exchange and also the telnet is working). The curl on the http port works as well: 

```curl --user weblogic:welcome1 -H X-Requested-By:MyClient -H Accept:application/json -X GET http://130.61.63.87:30701/management/weblogic```

So I tried also from the server in the kubernetes cluster and i faced the same issue.

The only way that worked was to go to the server running the Admin Container and try to connect to the pod IP : 

```python
wls:/offline> connect('weblogic','welcome1','t3://10.99.66.5:7001')                        
Connecting to t3://10.99.66.5:7001 with userid weblogic ...                                
Successfully connected to Admin Server "admin-server" that belongs to domain "base_domain".
                                                                                           
Warning: An insecure protocol was used to connect to the server.                           
To ensure on-the-wire security, the SSL port or Admin port should be used instead.         
                                                                                           
wls:/base_domain/serverConfig/>                                                            
```
Any cloue on what is wrong ?

**Comment by antonionappi88** [comment]
Hello,
about this the problem. It is caused by a wrong value in the `create-domain-job-input.yaml` for the t3PublicAddress. Probably it would be worth add to the [documentation](https://github.com/oracle/weblogic-kubernetes-operator/blob/master/site/creating-domain.md)  a reference and explain why it is needed and what should be the value. (the master ip or the LB's ip in front of the masters).

I tried to change this value in a running domain, WebLogic says that no restart is needed but actually it is not true. Only adding to the input file it is loaded (or restarting the admin).

**Comment by antonionappi88** [comment]
Hi @rjeberhard I have some updates.

In my OCI environment I have 6 nodes : 

- 2 etcd servers
- 2 master servers 
- 2 worker servers

Actually the NodePort is open on all nodes that are running kube-proxy (in OCI also on masters) so all the nodes in the K8S are able to redirect the external traffic to the correct pod but the masters don't run pods ( to be sure i shut-down the workers and  the scheduler was not able to find a valid node for the pod to be restarted.)

I tried to setup the `t3PublicAddress` to public ip of  a worker and taking care that the Admin was starting on that worker. 
The connection using WLST script was failing even i could telnet the port on the servers.

Instead when as `t3PublicAddress` I put the public ip of the LB in front of masters it worked.

As you said it is strange, any public ip where NodePort is enabled should work so I compared the OCI security list for the Master and Workers and I noticed  that adding this rule to the worker security list: 

```
Source: 10.0.0.0/16
IP Protocol: TCP
Source Port Range: All
Destination Port Range: 8080
Allows: TCP traffic for ports: 8080 
```
it worked. So also adding a public IP of LB in front of workers will work. 
The port 8080 is the port where the kube-apiserver is listening.

To resume the WLST script  against an OCI cluster will work if we : 
- use the public IP of the master's LB
- or we modify the terraform to open the communication on this port for the worker security list.

There is a things that is not clear to me. 
In the worker and master security lists I have also this rule : 

```
Source: 10.0.0.0/16
IP Protocol: All Protocols
Allows: all traffic for all ports
```
that it **should** open all the internal traffic on any port but you are forced to defined another rule for TCP to make it working 😕
